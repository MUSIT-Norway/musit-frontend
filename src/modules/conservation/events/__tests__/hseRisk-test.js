// @flow
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import HseRisk from '../hseRisk';

const props: any = {
  affectedThingsWithDetailsMainEvent: [
    {
      objectData: {
        id: 1,
        uuid: 'ad1b5287-0168-4343-b3df-a477de113bee',
        museumId: 99,
        museumNo: 'MusK58',
        subNo: '1',
        term: 'Rar dings',
        mainObjectId: 1,
        collection: 1,
        materials: [],
        locations: [],
        coordinates: [],
        isDeleted: false,
        objectType: 'collection'
      },
      id: 1,
      uuid: 'ad1b5287-0168-4343-b3df-a477de113bee',
      museumId: 99,
      museumNo: 'MusK58',
      subNo: '1',
      term: 'Rar dings',
      mainObjectId: 1,
      collection: 1,
      materials: [],
      locations: [],
      coordinates: [],
      isDeleted: false,
      objectType: 'collection'
    }
  ],
  hseRisk: {
    note: 'test',
    affectedThings: ['ad1b5287-0168-4343-b3df-a477de113bee']
  },
  index: 1,
  appSession: {
    accessToken: '123',
    actor: {
      fn: 'Rituvesh Kumar'
    },
    groups: [
      {
        id: 99,
        shortName: 'Test',
        museumId: 99,
        museumName: 'Test',
        permission: 10000,
        collections: [{ uuid: '00000000-0000-0000-0000-000000000000', name: 'All' }]
      }
    ],
    museumId: 99,
    collectionId: '00000000-0000-0000-0000-000000000000',
    buildInfo: {
      commitSha: 'not built on CI'
    },
    language: { isEn: true, isNo: false }
  },
  viewMode: true
};

describe('HseRisk', () => {
  it('should render properly', () => {
    const wrapper = shallow(<HseRisk {...props} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
