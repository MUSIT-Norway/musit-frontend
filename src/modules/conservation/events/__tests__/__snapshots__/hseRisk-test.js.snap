// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HseRisk should render properly 1`] = `
<CollapsibleEvent
  eventComponent={
    <div
      className="container"
    >
       
      <Component
        getDisplayNameForRole={[Function]}
        personData={Array []}
      />
       
      <div
        className="form-group"
      >
        <label
          className="control-label col-md-2"
          htmlFor="note_1"
        >
          Merknad:
        </label>
        <div
          className="col-md-9"
        >
          <textarea
            className="form-control"
            disabled={true}
            id="note_1"
            onChange={[Function]}
            rows="5"
            value="test"
          />
        </div>
      </div>
      <ObjectSelection
        affectedThingsSubEvent={
          Array [
            "ad1b5287-0168-4343-b3df-a477de113bee",
          ]
        }
        affectedThingsSubEventOnChange={[Function]}
        affectedThingsWithDetailsMainEvent={
          Array [
            Object {
              "collection": 1,
              "coordinates": Array [],
              "id": 1,
              "isDeleted": false,
              "locations": Array [],
              "mainObjectId": 1,
              "materials": Array [],
              "museumId": 99,
              "museumNo": "MusK58",
              "objectData": Object {
                "collection": 1,
                "coordinates": Array [],
                "id": 1,
                "isDeleted": false,
                "locations": Array [],
                "mainObjectId": 1,
                "materials": Array [],
                "museumId": 99,
                "museumNo": "MusK58",
                "objectType": "collection",
                "subNo": "1",
                "term": "Rar dings",
                "uuid": "ad1b5287-0168-4343-b3df-a477de113bee",
              },
              "objectType": "collection",
              "subNo": "1",
              "term": "Rar dings",
              "uuid": "ad1b5287-0168-4343-b3df-a477de113bee",
            },
          ]
        }
        viewMode={true}
      />
    </div>
  }
  eventName="HMS-risiko"
/>
`;
